src/ligero/mod.rs

```diff
@@ -0,0 +1,443 @@
+            sec_param: pp.sec_param,
+            rho_inv: pp.rho_inv,
+            leaf_hash_params: pp.leaf_hash_params.clone(),
+            two_to_one_params: pp.two_to_one_params.clone(),
+            check_well_formedness: pp.check_well_formedness,
+        };
+        Ok((ck, vk))
+    }
+
+    fn commit<'a>(
+        ck: &Self::CommitterKey,
+        polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F, P>>,
+        _rng: Option<&mut dyn RngCore>,
+    ) -> Result<
+        (
+            Vec<LabeledCommitment<Self::Commitment>>,
+            Vec<Self::Randomness>,
+        ),
+        Self::Error,
+    >
+    where
+        P: 'a,
+    {
+        let mut commitments = Vec::new();
+
+        for labeled_polynomial in polynomials.into_iter() {
+            let polynomial = labeled_polynomial.polynomial();
+
+            // 1. Arrange the coefficients of the polynomial into a matrix,
+            // and apply Reed-Solomon encoding to get `ext_mat`.
+            let (mat, ext_mat) = Self::compute_matrices(polynomial, ck.rho_inv);
+
+            // 2. Create the Merkle tree from the hashes of each column.
+            let col_tree =
+                Self::create_merkle_tree(&ext_mat, &ck.leaf_hash_params, &ck.two_to_one_params);
+
+            // 3. Obtain the MT root and add it to the transcript.
+            let root = col_tree.root();
+
+            let mut transcript: IOPTranscript<F> = IOPTranscript::new(b"transcript");
+
+            transcript
+                .append_serializable_element(b"root", &root)
+                .map_err(|_| Error::TranscriptError)?;
+
+            let n_rows = mat.n;
+            let n_cols = mat.m;
+            let n_ext_cols = ext_mat.m;
+
+            // 4. The commitment is just the root, but since each commitment could be to a differently-sized polynomial, we also add some metadata.
+            let commitment = LigeroPCCommitment {
+                metadata: Metadata {
+                    n_rows,
+                    n_cols,
+                    n_ext_cols,
+                },
+                root,
+            };
+
+            commitments.push(LabeledCommitment::new(
+                labeled_polynomial.label().clone(),
+                commitment,
+                None,
+            ));
+        }
+        let com_len = &commitments.len();
+        Ok((commitments, vec![Self::Randomness::empty(); *com_len]))
+    }
+
+    fn open<'a>(
+        ck: &Self::CommitterKey,
+        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F, P>>,
+        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
+        point: &'a P::Point,
+        _challenge_generator: &mut crate::challenge::ChallengeGenerator<F, S>,
+        _rands: impl IntoIterator<Item = &'a Self::Randomness>,
+        _rng: Option<&mut dyn RngCore>,
+    ) -> Result<Self::Proof, Self::Error>
+    where
+        P: 'a,
+        Self::Randomness: 'a,
+        Self::Commitment: 'a,
```

Comment:
Should we add `'static` bounds to `Self::Randomness` and `Self::Commitment` in the trait definition directly?